numpy/distutils/misc_util.py

```diff
@@ -92,7 +93,11 @@ def get_num_build_jobs():
 
     """
     from numpy.distutils.core import get_distribution
-    envjobs = int(os.environ.get("NPY_NUM_BUILD_JOBS", 1))
+    try:
+        cpu_count = len(os.sched_getaffinity(0))
+    except AttributeError:
+        cpu_count = multiprocessing.cpu_count()
```

Comment:
Is this safe? Is it always faster?

Reply:
Yes, it's safe and it's always faster (unless you're compiling with high CPU load and you have an OS scheduler that sucks, which is almost never the case nowadays).

Reply:
high CPU load = gaming AND compiling numpy at the same time (I don't play games).